test/parallel/test-module-isBuiltIn.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const { isBuiltIn } = require('module');
+
+// Includes modules in lib/ (even deprecated ones)
+assert(isBuiltIn('http'));
+assert(isBuiltIn('sys'));
+assert(isBuiltIn('node:fs'));
+assert(isBuiltIn('node:test'));
+
+// Does not include internal modules
+assert(!isBuiltIn('internal'));
```

Comment:
To make sure the function is properly checking for prefixes when required:

```suggestion
assert(!isBuiltIn('internal'));
assert(!isBuiltIn('test'));
```